internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -273,7 +273,11 @@ func copyRunDirectory(log *logger.Logger, newHash string) error {
 		log.Debugw("Run directory not present", "old_run_path", oldRunPath)
 		return nil
 	}
-	return errors.New(err, "failed to copy %q to %q", oldRunPath, newRunPath)
+	if err != nil {
```

Comment:
We could probably have a unit test for just this function that would have caught this problem.